### PR TITLE
feat: v2alpha user service idp endpoints

### PR DESCRIPTION
**Pull request by stebenz**
### Definition of Ready

- [ ] I am happy with the code
- [ ] Short description of the feature/issue is added in the pr description
- [ ] PR is linked to the corresponding user story
- [ ] Acceptance criteria are met
- [ ] All open todos and follow ups are defined in a new ticket and justified
- [ ] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [ ] No debug or dead code
- [ ] Critical parts are tested automatically
- [ ] Where possible E2E tests are implemented
- [ ] Documentation/examples are up-to-date
- [ ] All non-functional requirements are met
- [ ] Functionality of the acceptance criteria is checked manually on the dev system.

### CODE REVIEW

**Review by muhlemmer — CHANGES_REQUESTED** [review]
I let some comments in the proto files regarding paths. But I also noticed that we were not consistent with path names at all. Some examples:


https://github.com/zitadel/zitadel/blob/d19ed2258440e9d39ddc5607d90c22c56083ad23/proto/zitadel/user/v2alpha/user_service.proto#L84

https://github.com/zitadel/zitadel/blob/d19ed2258440e9d39ddc5607d90c22c56083ad23/proto/zitadel/user/v2alpha/user_service.proto#L162

https://github.com/zitadel/zitadel/blob/d19ed2258440e9d39ddc5607d90c22c56083ad23/proto/zitadel/user/v2alpha/user_service.proto#L230

File: internal/repository/intent/intent.go
[the remark this answers is not in the document]
Comment (stebenz) [reply internal/repository/intent/intent.go:1]: it was startet with intent and I guess idpintent was created to not intervene on the same branch, will merge both

File: internal/api/idp/idp.go [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+	return func(ctx context.Context) string {
+		return http_utils.BuildOrigin(authz.GetInstance(ctx).RequestedHost(), externalSecure) + HandlerPrefix + callbackPath
+	}
+}
+
+func NewHandler(
+	commands *command.Commands,
+	queries *query.Queries,
+	encryptionAlgorithm crypto.EncryptionAlgorithm,
+	externalSecure bool,
```
[the remark this answers is not in the document]
Comment (muhlemmer) [reply internal/api/idp/idp.go:65]: I thought `externalSecure` but I see it now.

File: internal/api/idp/idp.go [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+	case *openid.Provider:
+		session = &openid.Session{Provider: provider, Code: code}
+	case *azuread.Provider:
+		session = &oauth.Session{Provider: provider.Provider, Code: code}
+	case *github.Provider:
+		session = &oauth.Session{Provider: provider.Provider, Code: code}
+	case *gitlab.Provider:
+		session = &openid.Session{Provider: provider.Provider, Code: code}
+	case *google.Provider:
+		session = &openid.Session{Provider: provider.Provider, Code: code}
```
[the remark this answers is not in the document]
Comment (stebenz) [reply internal/api/idp/idp.go:203]: Have to correct my initial comment, the logic is there for ldap and jwt but placed somewhere else, so will create an error with a different ID but same message as the text is correct.

File: internal/command/user_human.go [hunk 3]
```diff
@@ -226,6 +235,13 @@ func (c *Commands) AddHumanCommand(human *AddHuman, orgID string, passwordAlg cr
 					metadataEntry.Value,
 				))
 			}
+			for _, link := range human.Links {
+				cmd, err := addLink(ctx, filter, a, link)
+				if err != nil {
+					return nil, err
+				}
+				cmds = append(cmds, cmd)
+			}
```
[the remark this answers is not in the document]
Comment (stebenz) [reply internal/command/user_human.go:244]: You're correct, but the problem is that currently there is no aggregateID for the IDPs, so if there is the information of orgID and instanceID this is the only way to select IDPs without directly scanning the data for the ID of the IDP.
We thought on changing that to the new writeModel, but currently not sure if that would be better.

[the remark this answers is not in the document]
Comment (muhlemmer) [reply internal/command/user_human.go:244]: I've created a follow up issue so we can discuss it with the team, perhaps during review or refinement.

**Review by muhlemmer — APPROVED** [review]

**Pull request merged**